libs/asset-viewer/src/lib/state/asset-search/asset-search.effects.ts

```diff
@@ -141,27 +154,37 @@ export class AssetSearchEffects {
     share()
   );
 
-  // noinspection JSUnusedGlobalSymbols
   public readSearchQueryParams$ = createEffect(() =>
     this.queryParams$.pipe(
-      filter(({ query }) => Object.values(query).some((value) => value !== undefined)),
-      map(({ query }) => actions.search({ query: query }))
+      withLatestFrom(this.store.select(selectAssetSearchQuery)),
+      filter(([params, query]) => !deepEqual(params.query, query)),
+      map(([params, storeQuery]) => {
```

Comment:
same here as below, i would prefer the same variables to have the same name in the entire pipe